server/src/main/java/org/opensearch/index/translog/RemoteTranslogStats.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.translog;
+
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentFragment;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.index.remote.RemoteTranslogTransferTracker;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Encapsulates the stats related to Remote Translog Store operations
+ *
+ * @opensearch.internal
+ */
+public class RemoteTranslogStats implements ToXContentFragment, Writeable {
+    /**
+     * Total number of Remote Translog Store uploads that have been started
+     */
+    private long totalUploadsStarted;
+
+    /**
+     * Total number of Remote Translog Store uploads that have failed.
+     */
+    private long totalUploadsFailed;
+
+    /**
+     * Total number of Remote Translog Store uploads that have been successful.
+     */
+    private long totalUploadsSucceeded;
+
+    /**
+     * Total number of byte uploads to Remote Translog Store that have been started.
+     */
+    private long uploadBytesStarted;
+
+    /**
+     * Total number of byte uploads to Remote Translog Store that have failed.
+     */
+    private long uploadBytesFailed;
+
+    /**
+     * Total number of byte uploads to Remote Translog Store that have been successful.
+     */
+    private long uploadBytesSucceeded;
+
+    static final String REMOTE_STORE = "remote_store";
+
+    static final String UPLOAD = "upload";
+
+    static final class SubFields {
+        static final String STARTED = "started";
+        static final String SUCCEEDED = "succeeded";
+        static final String FAILED = "failed";
+    }
+
+    static final class Fields {
+        static final String TOTAL_UPLOADS = "total_uploads";
+        static final String TOTAL_UPLOADS_IN_BYTES = "total_uploads_in_bytes";
```

Comment:
We should be able to reuse some of the static public fields already declared in `RemoteStoreStats.java`.

Reply:
Those fields are not public in `RemoteStoreStats.java`. Do you recommend we make them public? The package differs between these 2 classes.

Reply:
Made these public as per discussion over call. Resolving this thread. Please reopen if needed.